src/main/kotlin/com/github/mitchellolsthoorn/testgenie/toolwindow/TestGenieToolWindow.kt

```diff
@@ -0,0 +1,76 @@
+package com.github.mitchellolsthoorn.testgenie.toolwindow
+
+import com.intellij.openapi.ui.Messages
+import com.intellij.openapi.wm.ToolWindow
+import java.awt.Dimension
+import java.awt.event.ActionEvent
+import javax.swing.JButton
+import javax.swing.JComboBox
+import javax.swing.JComponent
+import javax.swing.JLabel
+import javax.swing.JPanel
+import javax.swing.JTextField
+
+/**
+ * This class stores the UI of the TestGenie tool window.
+ */
+class TestGenieToolWindow(_toolWindow: ToolWindow) {
+
+    private val toolWindow = _toolWindow
+
+    private var maxSizeTextField: JTextField? = null
+    private var globalTimeOutTextField: JTextField? = null
+    private var coverageCombobox: JComboBox<Boolean>? = null
+    private var saveButton: JButton? = null
+
+    private var toolWindowPanel: JPanel? = null
+
+    init {
+        saveButton?.addActionListener { addListenerForSaveButton(it) }
+    }
+
+    /**
+     * Returns the panel that is the main wrapper component of the tool window.
+     */
+    fun getContent(): JComponent? {
+        return toolWindowPanel!!
+    }
+
+
+    /**
+     * Adds a listener to the `Save` button to parse, validate and extract the entered values.
+     */
+    private val addListenerForSaveButton : (ActionEvent) -> Unit = {
+        if (maxSizeTextField?.text == null || globalTimeOutTextField?.text == null) {
+            Messages.showErrorDialog("Please specify the value", "Empty Value Field")
+        } else {
+            // Validate all the input values
+            val maxSize: Int? = toInt(maxSizeTextField!!.text)
+            val globalTimeout: Int? = toInt(globalTimeOutTextField!!.text)
+            val coverage : Boolean = coverageCombobox?.selectedItem.toString().toBoolean()
+
+            if (maxSize == null || globalTimeout == null) {
+                Messages.showErrorDialog("Please specify a number", "Invalid Input Value")
+            } else {
+                Messages.showInfoMessage(
+                    "Settings saved (not actually, still WIP)" +
+                            "\nMax size: $maxSize" +
```

Comment:
I think this is fine for now. But if I understand correctly we will have more parameters later on, maybe we should try to find a way or a design pattern that would avoid hard coding messages like that.

Reply:
Yes, that's a good idea. We discussed with Kiril that we might create something like a dynamic creation at runtime by reading EvoSuite class. Though, i don't know if it will introduce (much) overhead on the Settings page creation, since, essentially, the page is not created when the user opens the app, rather, it is created when the user selects the corresponding menu entry, and is destroyed when it is closed.

In either case, this will be addressed in the next sprint in a separate issue.